applications/fishing-map/src/features/map/popups/UserContextLayers.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { Fragment, useCallback } from 'react'
+// import { ContextLayerType } from '@globalfishingwatch/layer-composer/dist/generators/types'
+import groupBy from 'lodash/groupBy'
+import { batch, useSelector } from 'react-redux'
+import { useTranslation } from 'react-i18next'
+import IconButton from '@globalfishingwatch/ui-components/dist/icon-button'
+import { TooltipEventFeature } from 'features/map/map.hooks'
+import { useLocationConnect } from 'routes/routes.hook'
+import { CONTEXT_LAYER_PREFIX } from 'features/dataviews/dataviews.utils'
+import { selectHasAnalysisLayersVisible } from 'features/workspace/workspace.selectors'
+import styles from './Popup.module.css'
+
+type UserContextLayersProps = {
+  features: TooltipEventFeature[]
+  showFeaturesDetails: boolean
+}
+
+function ContextTooltipSection({ features, showFeaturesDetails = false }: UserContextLayersProps) {
+  const { t } = useTranslation()
+  const { dispatchQueryParams } = useLocationConnect()
+  const hasAnalysisLayers = useSelector(selectHasAnalysisLayersVisible)
+
+  const onReportClick = useCallback(
+    (feature: TooltipEventFeature) => {
+      if (!feature.properties?.gfw_id) {
+        console.warn('No gfw_id available in the feature to analyze', feature)
+        return
+      }
+
+      batch(() => {
+        dispatchQueryParams({
+          analysis: {
+            areaId: feature.properties?.gfw_id,
+            sourceId: feature.source,
+          },
+        })
+        // TODO decide if we keep the tooltip open or not
+        // dispatch(setClickedEvent(null))
+      })
+    },
+    [dispatchQueryParams]
+  )
+
+  const featuresByType = groupBy(features, 'layerId')
+  return (
+    <Fragment>
+      {Object.values(featuresByType).map((featureByType, index) => (
+        <div key={`${featureByType[0].title}-${index}`} className={styles.popupSection}>
+          <span
+            className={styles.popupSectionColor}
+            style={{ backgroundColor: featureByType[0].color }}
+          />
+          <div className={styles.popupSectionContent}>
+            {showFeaturesDetails && (
+              <h3 className={styles.popupSectionTitle}>{featureByType[0].title}</h3>
+            )}
+            {featureByType.map((feature) => {
+              const { gfw_id } = feature.properties
+              const isContextArea = feature.layerId.includes(CONTEXT_LAYER_PREFIX)
```

Comment:
User context layers generator is used for both (just contextual areas or environmental layers) so we need to distinguish between those two ones. Do you think worth it to move to another generator ? Because at the end it returns two completely different layers:
```
if (config.steps?.length && config.colorRamp) {... do Environmental...
else {
do regular context...
}
```

Reply:
I do, but maybe in a future refacto.
Seems to me like we'd be better off separating generators thinking more about cartography (`StaticPolygons`, `ChoroplethPolygons`, `Clusters`, etc) rather than whether it's user defined or not or if it represents context, environment, fishing etc